File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hash = {};

    for (let i = 0; i < nums.length; i++) {
        const partner = target - nums[i];

        if (partner in hash) {
            return [hash[partner], i];
        }
        hash[nums[i]] = i;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        
        for i, num in enumerate(nums):
            partner = target - num
            if partner in dict:
                return [i, dict[partner]]
            
            dict[num] = i